vaadin-list-mixin.html

```diff
@@ -208,18 +210,27 @@
       return -1;
     }
 
+    _setHidden(item) {
+      if (getComputedStyle(item).getPropertyValue('display') === 'none') {
+        item.hidden = true;
+      }
+    }
+
     _setFocusable(idx) {
-      idx = this._getAvailableIndex(idx, 1, item => !item.disabled);
+      idx = this._getAvailableIndex(idx, 1, item => !(item.disabled || getComputedStyle(item).getPropertyValue('display') === 'none'));
       const item = this.items[idx] || this.items[0];
       this.items.forEach(e => e.tabIndex = e === item ? 0 : -1);
     }
 
     _focus(idx) {
       const item = this.items[idx];
-      this.items.forEach(e => e.focused = e === item);
+      this._setHidden(item);
```

Comment:
This shouldn't be here. `hidden` shouldn't be modified by the component. Looks like `hidden` is also never removed if the `display: none` is removed from an item

Reply:
done